mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -72,97 +77,107 @@ public DefaultEntityFactory(ProcessorContext context) {
-
-    CqlNameGenerator cqlNameGenerator = buildCqlNameGenerator(classElement);
-    Set<String> transientProperties = getTransientPropertyNames(classElement);
-
+    Set<String> encounteredPropertyNames = Sets.newTreeSet();
     SortedMap<Integer, PropertyDefinition> partitionKey = new TreeMap<>();
     SortedMap<Integer, PropertyDefinition> clusteringColumns = new TreeMap<>();
     ImmutableList.Builder<PropertyDefinition> regularColumns = ImmutableList.builder();
     ImmutableList.Builder<PropertyDefinition> computedValues = ImmutableList.builder();
-    for (Element child : classElement.getEnclosedElements()) {
-      Set<Modifier> modifiers = child.getModifiers();
-      if (child.getKind() != ElementKind.METHOD
-          || modifiers.contains(Modifier.STATIC)
-          || modifiers.contains(Modifier.PRIVATE)) {
-        continue;
-      }
-      ExecutableElement getMethod = (ExecutableElement) child;
-      String getMethodName = getMethod.getSimpleName().toString();
-      if (!getMethodName.startsWith("get") || !getMethod.getParameters().isEmpty()) {
-        continue;
-      }
-      TypeMirror typeMirror = getMethod.getReturnType();
-      if (typeMirror.getKind() == TypeKind.VOID) {
-        continue;
-      }
-      String propertyName = Introspector.decapitalize(getMethodName.substring(3));
-      String setMethodName = getMethodName.replaceFirst("get", "set");
-      ExecutableElement setMethod = findSetMethod(classElement, setMethodName, typeMirror);
-      if (setMethod == null) {
-        continue; // must have both
-      }
-      VariableElement field = findField(classElement, propertyName, typeMirror);
 
-      Map<Class<? extends Annotation>, Annotation> propertyAnnotations =
-          scanPropertyAnnotations(getMethod, field);
-      if (isTransient(propertyAnnotations, propertyName, transientProperties, getMethod, field)) {
-        continue;
-      }
+    // scan hierarchy for properties
+    for (TypeElement typeElement : typeHierarchy) {
+      for (Element child : typeElement.getEnclosedElements()) {
+        Set<Modifier> modifiers = child.getModifiers();
+        if (child.getKind() != ElementKind.METHOD
+            || modifiers.contains(Modifier.STATIC)
+            || modifiers.contains(Modifier.PRIVATE)) {
+          continue;
+        }
+        ExecutableElement getMethod = (ExecutableElement) child;
+        String getMethodName = getMethod.getSimpleName().toString();
+        if (!getMethodName.startsWith("get") || !getMethod.getParameters().isEmpty()) {
+          continue;
+        }
+        TypeMirror typeMirror = getMethod.getReturnType();
+        if (typeMirror.getKind() == TypeKind.VOID) {
+          continue;
+        }
+        String propertyName = Introspector.decapitalize(getMethodName.substring(3));
+        // skip properties we've already encountered.
+        if (encounteredPropertyNames.contains(propertyName)) {
+          continue;
+        }
```

Comment:
If I read `HierarchyScanner` correctly, `typeHierarchy` is in ascending order, so we'll run into child classes first. This could be a problem if we want our annotations to be inherited, for example:
```java
interface Identifiable {
  @PrimaryKey UUID getId();
  void setId(UUID id);
}
@Entity
class Product implements Identifiable {
  @Override public UUID getId() { ... }
  @Override public void setId(UUID id) { ... }

}
```
Maybe making our annotations `@Inherited` would be enough to address this, but it's probably not resolved when parsing annotation mirrors.

Another approach could be to traverse the set in reverse order. But that would prevent a class from overriding a method with additional annotations.

In any case we need to define rules for annotation inheritance, write them down in the manual and test them.

Reply:
Actually I just realized you're handling inheritance in `AnnotationScanner#getMethodAnnotation`, correct?

Reply:
That's right, it works similarly to the 3.x mapper, parsing out the properties is done separately from parsing the annotations.  Once you've identified the properties, you find the annotations, which can be pulled from different levels of the hierarchy.

Reply:
Although that is only the case for method annotations.  We only pull annotations from the identified field (also similar to 3.x mapper)

Reply:
Sounds good, fields can't be overridden anyway so it's unlikely that the same one could be present at different levels.